modules/pipeline/pipengine/reconciler/pipeline_gc_compensator.go

```diff
@@ -89,7 +90,7 @@ func (r *Reconciler) getNeedGCPipelines(pageNum int, isSnippet bool) ([]spec.Pip
 				ttl = defaultGCTime
 			}
 
-			if p.TimeEnd.IsZero() {
+			if p.TimeEnd == nil || p.TimeEnd.IsZero() {
```

Comment:
if TimeEnd is empty and status is endStatus, you can use TimeUpdate field.
Otherwise, these namespaces will not be gc.